profiles/profile.py

```diff
@@ -1,155 +1,161 @@
+        )
+        plate_map_name = barcode_plate_map_df.query(
+            "Assay_Plate_Barcode == @plate"
+        ).Plate_Map_Name.values[0]
+        plate_map_file = pathlib.PurePath(metadata_dir, "platemap", f"{plate_map_name}.txt")
+        plate_map_df = pd.read_csv(plate_map_file, sep="\t")
+        plate_map_df.columns = [
+            f"Metadata_{x}" if not x.startswith("Metadata_") else x
+            for x in plate_map_df.columns
+        ]
+        platemap_well_column = pipeline["platemap_well_column"]
+
+        if annotate_steps["external"]:
+            external_df = pd.read_csv(
+                pathlib.PurePath(".", "metadata", "moa", annotate_steps["external"]),
+                sep="\t",
+            )
+            anno_df = annotate(
+                profiles=aggregate_output_file,
+                platemap=plate_map_df,
+                join_on=[platemap_well_column, annotate_well_column],
+                cell_id=cell,
+                external_metadata=external_df,
+                external_join_left=["Metadata_broad_sample"],
+                external_join_right=["Metadata_broad_sample"],
+            )
+        else:
+            anno_df = annotate(
+                profiles=aggregate_output_file,
+                platemap=plate_map_df,
+                join_on=[platemap_well_column, annotate_well_column],
+                cell_id=cell,
+            )
+
+        anno_df = anno_df.rename(
+            {
+                "Image_Metadata_Plate": "Metadata_Plate",
+                "Image_Metadata_Well": "Metadata_Well",
+            },
+            axis="columns",
+        ).assign(
+            Metadata_Assay_Plate_Barcode=plate,
+            Metadata_Plate_Map_Name=barcode_plate_map_df.loc[
+                barcode_plate_map_df.Assay_Plate_Barcode == plate, "Plate_Map_Name"
+            ].values[0],
+        )
+
+        # Reoroder columns
+        metadata_cols = cyto_utils.infer_cp_features(anno_df, metadata=True)
+        cp_cols = cyto_utils.infer_cp_features(anno_df)
+        reindex_cols = metadata_cols + cp_cols
+        anno_df = anno_df.reindex(reindex_cols, axis="columns")
+
+        # Output annotated file
+        cyto_utils.output(
+            df=anno_df,
+            output_filename=annotate_output_file,
+            float_format=float_format,
+            compression=compression,
+        )
+
+    # Normalize Profiles
+    normalize_steps = pipeline["normalize"]
+    if normalize_steps["perform"]:
+        normalization_features = normalize_steps["features"]
+        normalization_method = normalize_steps["method"]
+        normalize(
+            profiles=annotate_output_file,
+            features=normalization_features,
+            samples=samples,
+            method=normalization_method,
+            output_file=normalize_output_file,
+            float_format=float_format,
+            compression=compression,
+        )
+        if normalize_steps["negcon"]:
```

Comment:
I don't think that we'll have a good solution to support a block execution design before we merge this PR.

As an intermediate and sustainable step, I propose the following:

1. Drop the `normalize_steps["negcon"]` (and the `feature_select_steps["negcon"]`) option
2. Create two config yaml files (one with normalize -> samples set to `"Metadata_control_type == 'negcon'"`, and the other with normalize -> samples set to "all"). We can avoid rerunning aggregation by setting perform = False in the second executed config file
3. Extract out [line 23](https://github.com/jump-cellpainting/profiling-recipe/pull/14/files#diff-b9d876056b6394f85313599f3b2cf8500f517537c9a8f519afb74c6b13712f0aR23) that set filename suffixes into the configuration file
4. Add functionality to `profiles/profile_utils.py` in order to parse the new file names in the config file